tests/ui/on-unimplemented/no-debug.stderr

```diff
@@ -17,7 +19,9 @@ error[E0277]: `Bar` doesn't implement `Debug`
   --> $DIR/no-debug.rs:10:32
    |
 LL |     println!("{:?} {:?}", Foo, Bar);
-   |                                ^^^ `Bar` cannot be formatted using `{:?}` because it doesn't implement `Debug`
+   |                    ----        ^^^ `Bar` cannot be formatted using `{:?}` because it doesn't implement `Debug`
```

Comment:
This change on diagnostic reminds me of this issue:
https://github.com/rust-lang/rust/issues/140578

while we had a PR for it, but seems it's stalled.

Will this PR also resolve #140578? If so, we can link with it, add a unit test, and close that one too.

Reply:
It doesn't, but it does make that PR easier to implement; they can simply replace all of the diagnostic item stuff with a `matches!(span.desugaring_kind(), Some(rustc_span::DesugaringKind::FormatLiteral {..}))`